src/tsdb/inc/tsdbFile.h

```diff
@@ -257,7 +257,13 @@ static FORCE_INLINE int tsdbAppendDFile(SDFile* pDFile, void* buf, int64_t nbyte
     return -1;
   }
 
-  ASSERT(pDFile->info.size == toffset);
+  //bug fix. To avoid data corruption, close the data file in advance.
+  //ASSERT(pDFile->info.size == toffset);
+  if (pDFile->info.size != toffset) {
+    tsdbCloseDFile(pDFile);
```

Comment:
Sure to close pDFile here?

Reply:
Good comments. Normally the file should only be closed by the one who open it.

Reply:
Thanks. Fixed and committed again.